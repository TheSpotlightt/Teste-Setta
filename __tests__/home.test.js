import 'react-native';
import React from 'react';
import Home from '../src/screens/home';
import { act } from 'react-dom/test-utils';
import { render, fireEvent, waitFor } from '@testing-library/react-native';

import 'react-test-renderer';

test('should render home correctly', () => {
    render(<Home />);
});


test('should render title', () => {
    const { getByTestId } = render(<Home />);

    expect(getByTestId('title').props.children).toBe(' Star Wars Characters ');

});
