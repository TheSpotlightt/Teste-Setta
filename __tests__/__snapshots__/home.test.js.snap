// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render home 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#ffe81f",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginTop": 60,
            "textAlign": "center",
          },
        ]
      }
      testID="title"
    >
       Star Wars Characters 
    </Text>
    <RCTScrollView
      style={
        Array [
          Object {},
        ]
      }
    >
      <View>
        <ActivityIndicator
          animating={true}
          color="#ffe81f"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`should render title 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#ffe81f",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginTop": 60,
            "textAlign": "center",
          },
        ]
      }
      testID="title"
    >
       Star Wars Characters 
    </Text>
    <RCTScrollView
      style={
        Array [
          Object {},
        ]
      }
    >
      <View>
        <ActivityIndicator
          animating={true}
          color="#ffe81f"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
